crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -503,19 +518,38 @@ impl Timeline {
     ) -> Result<bool, RoomSendQueueError> {
         let event_id = match edit_info.id {
             TimelineEventItemId::TransactionId(txn_id) => {
-                let Some(item) = self.item_by_transaction_id(&txn_id).await else {
-                    warn!("Couldn't find the local echo anymore");
-                    return Ok(false);
-                };
+                if let Some(item) = self.item_by_transaction_id(&txn_id).await {
+                    let Some(handle) = item.as_local().and_then(|item| item.send_handle.clone())
+                    else {
+                        warn!("No handle for a local echo; is this a test?");
+                        return Ok(false);
+                    };
 
-                if let Some(handle) = item.as_local().and_then(|item| item.send_handle.clone()) {
                     // Assume no relations, since it's not been sent yet.
-                    let new_content: RoomMessageEventContent = new_content.into();
-                    return Ok(handle.edit(new_content.into()).await?);
+                    let new_content: RoomMessageEventContent = new_content.clone().into();
+
+                    if handle.edit(new_content.into()).await? {
+                        return Ok(true);
+                    }
                 }
 
-                warn!("No handle for a local echo; should only happen in testing situations");
-                return Ok(false);
+                // We end up here in two cases: either there wasn't a local echo with this
+                // transaction id, or the send queue refused to edit the local echo (likely
+                // because it's sent).
+                //
+                // Try to find a matching local echo that now has an event id (it's been sent),
+                // or a remote echo with a matching transaction id, so as to
+                // send an actual redaction.
```

Comment:
A redaction? We aren't redacting an event here, did you mean edit or replacement event?

Reply:
Indeedy, thanks! (wow you actually read the code comments :exploding_head:)

Reply:
There have been plenty times where I did not and have missed things, so I try to read them nowadays.